editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/restconfiguration/RestConfigEditor.java

```diff
@@ -694,4 +793,72 @@ public void handleEvent(Event event) {
 		}
 	}
 	
+	/**
+	 * Public for tests.
+	 */
+	public void addRestElement() {
+		Display.getDefault().syncExec(() -> {
+			RestElement newre = util.createRestElementNode(ctx.getCamelFile());
+			newre.initialize();
+			ctx.addRestElement(newre);
+			ctx.getCamelFile().fireModelChanged();
+			reload();
+		});
+	}
+
+	/**
+	 * Public for tests.
+	 */
+	public void addRestConfigurationElement() {
+		Display.getDefault().syncExec(() -> {
+			RestConfigurationElement newrce = util.createRestConfigurationNode(ctx.getCamelFile());
+			newrce.initialize();
+			newrce.setHost("localhost");
+			newrce.setBindingMode(OFF);
+			if (ctx.getRestConfigurations().isEmpty()) {
+				ctx.addRestConfiguration(newrce);
+				ctx.getCamelFile().fireModelChanged();
```

Comment:
I am pretty sure that this is not required as the xml document is already observed for changes in the dom tree and will do all needed event triggering.